mne/preprocessing/artifact_detection.py

```diff
@@ -4,12 +4,105 @@
 
 
 import numpy as np
-
 from ..annotations import (Annotations, _annotations_starts_stops)
 from ..transforms import (quat_to_rot, _average_quats, _angle_between_quats,
                           apply_trans, _quat_to_affine)
+from ..filter import filter_data
 from .. import Transform
-from ..utils import (_mask_to_onsets_offsets, logger)
+from ..utils import (_mask_to_onsets_offsets, logger, verbose)
+
+
+@verbose
+def annotate_muscle_zscore(raw, threshold=4, picks=None, min_length_good=0.1,
+                           filter_freq=(110, 140), n_jobs=1, verbose=None):
+    """Detect segments that likely contain muscle artifacts.
+
+    Detects data segments containing activity in the frequency range given by
+    ``filter_freq``, that exceeds the specified z-score threshold.
+
+    Raw data is band-pass filtered between ``filter_freq`` especified
+    frequencies (default is 110 - 140 Hz), the signal envelope computed,
+    z-scored across samples, channel summation and division by the square root
+    of the channel number, and low-pass filtered to better capture beginning
+    and end of muscle activity and false positive transient peaks.
+
+    .. note::
+        Use a single channel type.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to estimate segments with muscle artifacts.
+    threshold : float
+        The threshold in z-scores for marking segments as containing muscle
+        activity artifacts.
+    %(picks_all)s
+    min_length_good : float | None
+        The shortest allowed duration of "good data" (in seconds) between
+        adjacent annotations; shorter segments will be incorporated into the
+        surrounding annotations.``None`` is equivalent to ``0``.
+        Default is ``0.1``.
+    filter_freq : array-like, shape (2,)
+        The lower and upper frequencies of the band-pass filter.
+        Default is ``(110, 140)``.
+    %(n_jobs)s
+    %(verbose)s
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with muscle artifacts annotated as BAD_muscle.
+    scores_muscle : array
+        Z-score values averaged across channels for each sample.
+    """
+    from scipy.stats import zscore
+    from scipy.ndimage.measurements import label
+
+    raw_copy = raw.copy()
+    raw_copy.pick(picks)
+    ch_type = raw_copy.get_channel_types()
+
+    # Remove ref chans if MEG data just in case
+    meg = [True for e in ['mag', 'grad'] if (e in ch_type)]
+    if bool(meg) is True:
```

Comment:
This is better. I think even clearer would be something like:
```suggestion
    meg = any(np.in1d(ch_type, ('mag', 'grad')))
    if meg:
```

Reply:
I changed that, now ``ch_type`` is a parameter